Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -67,6 +67,40 @@ struct TextureMapperLayer::ComputeTransformData {
     }
 };
 
+class TextureMapperFlattenedLayer final {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    explicit TextureMapperFlattenedLayer(const IntRect& rect)
+        : m_rect(rect)
+    { }
+
+    void bind(TextureMapper& textureMapper)
+    {
+        if (!m_texture)
+            m_texture = textureMapper.acquireTextureFromPool(m_rect.size(), { BitmapTexture::Flags::SupportsAlpha });
+        textureMapper.bindSurface(m_texture.get());
+    }
+
+    inline IntRect layerRect() const
+    {
+        return m_rect;
+    }
+
+    RefPtr<BitmapTexture> surface() const
+    {
+        return m_texture;
+    }
+
+    void paintToTextureMapper(TextureMapper& textureMapper, const FloatRect& targetRect, const TransformationMatrix& modelViewMatrix, float opacity)
+    {
+        textureMapper.drawTexture(*m_texture.get(), targetRect, modelViewMatrix, opacity);
+    }
+
+private:
+    IntRect m_rect;
```

Comment:
BitmapTexture has size. I think you can remove m_rect.

Reply:
flattened layer may exceed the bounds of host layer in every direction not only size but x,y position also